accumulator/pollardutil.go

```diff
@@ -9,8 +9,9 @@ import (
 
 // PolNode is a node in the pollard forest
 type polNode struct {
-	data  Hash
-	niece [2]*polNode
+	data     Hash
+	niece    [2]*polNode
+	remember bool
```

Comment:
maybe creating a `polLeaf` type would be good because the remember bool is stored on all nodes now but really only the leaves need it.
Both `polNode` and `polLeaf` would need to inherit from the same interface.

Reply:
Hmm yeah. I also hate the fact that `niece` isn't nieces for roots and leaves. Maybe we can solve all these problems at once.